### ISSUE TITLE
Intermittent test failure on win32 (test_policies.ThrottlingTestCase.testLimit)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11186]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)| @teratorn reported|
|-|-|
|Trac ID|trac#1692|
|Type|defect|
|Created|2006-05-02 12:42:49Z|

The test, test_policies.ThrottlingTestCase.testLimit, fails intermittently on win32. For me, it is usually reproduceable with "trial -u" in about 5 passes.

The problem is with the test case, not the code under test.

The test case initiates 3 TCP connections (connectTCP) back-to-back, and relies on the corresponding connectionMade methods to be called in the same order. This does not always happen, at least not on win32.

The test case tests the ability of t.protocols.policies.ThrottlingFactory to drop a 3rd connection after 2 connections have already been made.

The most naive solution is to introduce a small waiting period before the 2nd and 3rd call to connectTCP (e.g. callLater with a value of 0.01, and 0.02 respectively)

FireMoth, I'm CC'ing you for review unless someone checks off on it sooner :)

<details><summary>Searchable metadata</summary>

```
trac-id__1692 1692
type__defect defect
reporter__teratorn teratorn
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__test test
time__1146573769000000 1146573769000000
changetime__1146634271000000 1146634271000000
version__ 
owner__ 
cc__FireMoth
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to FireMoth|
|-|-|

Fix as per above in branch testLimit-win32-1692

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn commented|
|-|-|

After further consideration, a time-based test is almost always a bad thing.

Will implement a non-time-based fix later.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to @teratorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to FireMoth|
|-|-|

OK. Fixed in a sensible way now. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to Jerub|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @jerub<br>@jerub set status to `assigned`|
|-|-|

approved.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @teratorn<br>@jerub set status to `new`|
|-|-|

please merge.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set status to `closed`|
|-|-|

(In [16723]) Merge testLimit-win32-1692
 
Author: teratorn
Review: Jerub
Fixes [#11186](https://github.com/twisted/twisted/issues/11186)

Ensure that our connectionMade methods are called in the correct order.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
